pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -50,8 +51,9 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 		isCopyOfBackupsRequired bool
 		tasksWithErrors         []string
 
-		isRestoring   = operationType == gardencorev1beta1.LastOperationTypeRestore
-		skipReadiness = metav1.HasAnnotation(o.Shoot.GetInfo().ObjectMeta, v1beta1constants.AnnotationShootSkipReadiness)
+		isRestoring           = operationType == gardencorev1beta1.LastOperationTypeRestore
+		skipReadiness         = metav1.HasAnnotation(o.Shoot.GetInfo().ObjectMeta, v1beta1constants.AnnotationShootSkipReadiness)
+		k8sGreaterEqual133, _ = versionutils.CheckVersionMeetsConstraint(o.Shoot.GetInfo().Spec.Kubernetes.Version, ">= 1.33")
```

Comment:
I think we should document this somewhere. We don't have a good place for it yet, but such things aren't obvious for people until they upgrade and run into it. I guess it makes sense to start a new document in https://github.com/gardener/gardener/tree/master/docs/usage/shoot where we describe for each Kubernetes version that we support what the differences/requirements to update are.